src/repo/repository.rs

```diff
@@ -115,6 +116,10 @@ pub struct ExtractResult {
 pub struct Repository {
     /// The path containing the [`Repository::data_dir`] directory.
     path: PathBuf,
+    /// A helper which handles the recursive removal of empty directories. The
+    /// queue is updated by calls to [`Repository::remove_extracted_file`] and
+    /// must be processed by calling [`EmptyDirectoryCleanupQueue::process()`].
+    dir_removal_queue: EmptyDirectoryCleanupQueue,
```

Comment:
Does this need to share its lifetime with the Repo? I might rather have expected to see this as a local variable which brackets the operation of doing filesystem updates.

Reply:
All right, I will make it local to the extract method.